day2/q3.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+var mu sync.Mutex
```

Comment:
instead of declaring mutex like this, we can have a instance level variable for mutex. 
example
```
type Account struct {
	balance int
        mu sync.Mutex
}
```

benefit: each instance of the Account struct has its own mutex, providing a level of isolation for the concurrent access to the balance field

Reply:
Noted . This makes more sense to me now. I thought global level declaration would have been better here but this seems more logical now.